Fails on some architectures: src/SendReportDialog.cpp:435:20: error: invalid output constraint '=a' in asm

On armv7 the error is:
```
src/SendReportDialog.cpp:435:20: error: invalid output constraint '=a' in asm
                 : "=a"(regs[0]), "=b"(regs[1]), "=c"(regs[2]), "=d"(regs[3])
                   ^
1 error generated.
```

If assembly is only for x86/x86_64 please place the check for architecture into cmake scripts.

--- a/src/ugenem/src/SendReportDialog.cpp
+++ b/src/ugenem/src/SendReportDialog.cpp
@@ -51,7 +51,7 @@
 #include "getMemorySize.c"
 
 #define HOST_URL "http://ugene.net"
-//#define HOST_URL "http://127.0.0.1:80"
+// #define HOST_URL "http://127.0.0.1:80"
 #ifdef Q_OS_LINUX
 #    define DESTINATION_URL_KEEPER_PAGE "/crash_reports_dest_breakpad_lin.html"
 #elif defined(Q_OS_DARWIN)
@@ -425,8 +425,12 @@ int ReportSender::getTotalPhysicalMemory() {
     return (int)(getMemorySize() / (1024 * 1024));
 }
 
-#ifndef Q_OS_DARWIN
-void cpuID(unsigned i, unsigned regs[4]) {
+#if !defined(Q_OS_DARWIN) && defined Q_PROCESSOR_X86
+#    define UGENE_HAS_CPU_ID
+#endif
+
+#ifdef UGENE_HAS_CPU_ID
+static void cpuID(unsigned i, unsigned regs[4]) {
 #    ifdef _WIN32
     __cpuid((int*)regs, (int)i);
 
@@ -441,7 +445,7 @@ void cpuID(unsigned i, unsigned regs[4]) {
 
 QString ReportSender::getCPUInfo() {
     QString result;
-#ifndef Q_OS_DARWIN
+#ifdef UGENE_HAS_CPU_ID
     unsigned regs[4];
 
     // Get vendor
@@ -492,5 +496,15 @@ void ReportSender::setFailedTest(const QString& failedTestStr) {
 }
 
 QString ReportSender::getArchSuffix() const {
+#ifdef Q_PROCESSOR_X86_64
     return " x64";
+#elif Q_PROCESSOR_X86_32
+    return " x86";
+#elif Q_PROCESSOR_ARM_64
+    return " arm-64";
+#elif Q_PROCESSOR_ARM_32
+    return " arm-32";
+#else
+    return " unknown-arch";
+#endif
 }

--- a/src/ugenem/src/SendReportDialog.cpp
+++ b/src/ugenem/src/SendReportDialog.cpp
@@ -51,7 +51,6 @@
 #include "getMemorySize.c"
 
 #define HOST_URL "http://ugene.net"
-// #define HOST_URL "http://127.0.0.1:80"
 #ifdef Q_OS_LINUX
 #    define DESTINATION_URL_KEEPER_PAGE "/crash_reports_dest_breakpad_lin.html"
 #elif defined(Q_OS_DARWIN)

